doc/modules/optim.rst

```diff
@@ -161,33 +161,228 @@ Model classes can be used with any solver class, by simply passing them using th
 solver's ``set_model`` method, see the :ref:`example given above <optim-first-example>`.
 
 
-.. _optim-model-glm:
+.. _linear-models:
 
-1.2. Generalized linear models
-------------------------------
+1.2. Linear models
+------------------
 
-We describe here generalized linear models for supervised learning.
+We describe here (generalized) linear methods for supervised learning.
```

Comment:
Well, because these are not really "models", for instance there is not model associated to the hinge loss... A big refactoring of model names and organization should be done maybe in another pull request (since involves refactoring and moving other models)